synapse/storage/databases/main/stream.py

```diff
@@ -1071,10 +1071,11 @@ def get_all_new_events_stream_txn(
             "get_all_new_events_stream", get_all_new_events_stream_txn
         )
 
-        events = await self.get_events_as_list(
-            event_to_received_ts.keys(),
-            get_prev_content=get_prev_content,
+        event_entries = await self.get_events_from_cache_or_db(
+            event_to_received_ts.keys(), get_prev_content=get_prev_content
         )
+        eventCache_list = list(event_entries.values())
+        events = [e.event for e in eventCache_list]
```

Comment:
This looks a bit odd---we end up building a whole list on 1077 only to throw it away inside 1078's comprehension.
How about:

```suggestion
        events = [e.event for e in event_entries.values()]
```